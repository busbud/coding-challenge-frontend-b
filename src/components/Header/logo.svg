<svg id="logo-busbud-white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 216 72.473"><path d="M50.453 22.437l-20.91-20.91a5.078 5.078 0 00-7.18-.005L1.485 22.4a5.077 5.077 0 00.005 7.18l6.69 6.69 5.18 5.18 6.926 6.927 2.113 2.113a5.147 5.147 0 00.63.529v18.35a3.03 3.03 0 106.06 0V53.198a5.536 5.536 0 01-3.091 1.109 4.563 4.563 0 01-2.968-1.098v-2.19a5.114 5.114 0 006.55-.524l2.255-2.254 6.79-6.791 4.734-4.735 7.1-7.099a5.078 5.078 0 00-.006-7.18" fill="#fbae17"></path><path d="M204.467 34.36s4.928-.505 4.928 9.466-4.803 9.88-4.803 9.88-5.076 1.05-5.076-9.743 4.951-9.602 4.951-9.602m-2.356 24.404a7.92 7.92 0 007.295-4.631v4.002H216V17.24h-6.594l.01 16.36c0-.675-2.533-4.37-7.15-4.37-6.592 0-9.69 6.134-9.69 15.464 0 15.322 9.535 14.07 9.535 14.07M173.589 58.748a8.667 8.667 0 008.498-6.038v5.429h6.59V29.912h-6.586V42.18c0 5.1-2.562 10.428-6.701 10.428-3.019 0-2.853-6.014-2.853-6.014V29.912h-6.556V49.28c0 3.104.765 9.467 7.608 9.467M150.127 33.443c2.397-.045 4.867 1.464 4.867 10.246s-2.28 10.611-4.867 10.611c-4.006 0-5.332-3.45-5.332-10.438s2.607-10.42 5.332-10.42m-.01 25.322c7.418 0 11.866-3.881 11.866-14.82s-4.21-14.732-9.65-14.732a8.273 8.273 0 00-7.464 4.244v-16.23h-6.599v29.07c0 8.095 4.43 12.468 11.847 12.468M75.184 33.443c2.398-.045 4.868 1.464 4.868 10.246S77.772 54.3 75.184 54.3c-4.005 0-5.331-3.45-5.331-10.438s2.607-10.42 5.332-10.42m-.01 25.322c7.418 0 11.867-3.881 11.867-14.82s-4.21-14.732-9.651-14.732a8.274 8.274 0 00-7.464 4.244v-16.23h-6.6v29.07c0 8.095 4.43 12.468 11.848 12.468M98.673 58.748a8.667 8.667 0 008.498-6.038v5.429h6.589V29.912h-6.586V42.18c0 5.1-2.561 10.428-6.7 10.428-3.019 0-2.853-6.014-2.853-6.014V29.912h-6.557V49.28c0 3.104.765 9.467 7.609 9.467M135.687 50.314c0-7.654-12.37-10.643-12.37-13.875 0-1.41 2.553-2.503 4.472-2.503 2.721 0 6.148 1.968 6.148 1.968V30.34a23.757 23.757 0 00-6.27-1.118c-2.43 0-10.158.729-10.158 8.214 0 7.679 12.102 10.692 12.102 13.852 0 1.215-1.19 2.405-5.103 2.405a16.363 16.363 0 01-7.096-1.75v5.638a24.196 24.196 0 007.582 1.167c4.398 0 10.693-1.385 10.693-8.433" fill="#fff"></path><path d="M37.61 24.137a11.745 11.745 0 01-23.46 0h-3.226a14.848 14.848 0 0029.692 0zM29.09 50.921a5.075 5.075 0 01-6.06.1v2.189a4.564 4.564 0 002.967 1.098 5.54 5.54 0 003.093-1.11v-2.276" fill="#e88623"></path></svg>
