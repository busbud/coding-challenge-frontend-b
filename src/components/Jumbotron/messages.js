import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'containers.jumbotron.title',
    defaultMessage: 'Find and compare cheap bus',
  },
});
