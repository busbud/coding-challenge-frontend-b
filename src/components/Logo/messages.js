import { defineMessages } from 'react-intl';

export default defineMessages({
  alt: {
    id: 'components.logo.alt',
    defaultMessage: 'Busbud',
  },
});
