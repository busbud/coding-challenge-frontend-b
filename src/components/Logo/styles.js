export default {
  root: {
    height: '100%',
  },
};
