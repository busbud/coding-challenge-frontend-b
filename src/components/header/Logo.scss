.logo--inline {
  display: flex;
  align-items: center;
}

.logo__img {
  margin-left: 10px;
  @media screen and (max-width: 600px) {
    width: 100px;
  }
}
