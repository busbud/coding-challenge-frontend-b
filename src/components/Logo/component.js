import React from 'react';
import { object } from 'prop-types';
import { intlShape } from 'react-intl';
import logo from './logo.svg';
import messages from './messages';

const Component = ({ intl: { formatMessage }, classes }) => (
  <img className={classes.root} alt={formatMessage(messages.alt)} src={logo} />
);

Component.propTypes = {
  intl: intlShape,
  classes: object,
};

export default Component;
