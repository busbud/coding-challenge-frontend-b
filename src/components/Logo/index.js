import { compose } from 'recompose';
import { withStyles } from 'material-ui/styles';
import { injectIntl } from 'react-intl';
import styles from './styles';
import component from './component';

export const enhance = compose(injectIntl, withStyles(styles));

export default enhance(component);
