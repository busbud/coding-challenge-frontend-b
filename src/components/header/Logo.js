import React from "react";
// Third party libraries
import counterpart from "counterpart";
import Translate from "react-translate-component";
import Typography from "@material-ui/core/Typography";
// Inner imports
import "./Logo.css";
import logo from "./img/oshega_xs.png";

const Logo = () => (
  <div className="logo--inline">
    <Typography variant="title" color="inherit">
      <Translate content="header.logo.text_before" />
    </Typography>
    <img
      className="logo__img"
      src={logo}
      alt={counterpart.translate("header.logo.alt")}
    />
  </div>
);

export default Logo;
