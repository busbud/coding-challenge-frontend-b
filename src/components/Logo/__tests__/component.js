import React from 'react';
import { shallowWithIntl } from '../../../utils/react-intl-test';
import Logo from '../component';
import logo from '../logo.svg';
import messages from '../messages';

const props = {
  classes: {
    root: 'fakeRootClass',
  },
};

describe('component | Logo | component', () => {
  describe('render', () => {
    it('should render without throwing and error', () => {
      // given
      const wrapper = shallowWithIntl(<Logo {...props} />);

      // then
      expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
      // given
      const wrapper = shallowWithIntl(<Logo {...props} />);

      // then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('img', () => {
    it('should have proper className props', () => {
      // given
      const wrapper = shallowWithIntl(<Logo {...props} />);
      const className = wrapper.props().className;

      // then
      expect(className).toBe(props.classes.root);
    });
    it('should have proper src props', () => {
      // given
      const wrapper = shallowWithIntl(<Logo {...props} />);
      const src = wrapper.props().src;

      // then
      expect(src).toBe(logo);
    });
    it('should have proper alt props', () => {
      // given
      const wrapper = shallowWithIntl(<Logo {...props} />);
      const alt = wrapper.props().alt;

      // then
      expect(alt).toBe(messages.alt.defaultMessage);
    });
  });
});
