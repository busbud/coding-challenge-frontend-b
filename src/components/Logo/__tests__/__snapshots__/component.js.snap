// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | Logo | component render should match snapshot 1`] = `
<img
  alt="Busbud"
  className="fakeRootClass"
  src="logo.svg"
/>
`;
